### PR TITLE
convert vit_small_ds Vision tutorial to Keras3

### PR DESCRIPTION
**Pull request by SuryanarayanaY**
I have done the changes to the Vision tutorial `vit_small_ds` to make suitable for Keras3.

The git diff file is attached below.
[git_diff_vit-small-ds.txt](https://github.com/keras-team/keras-io/files/13659000/git_diff_vit-small-ds.txt)

The changes tested and attached colab [gist](https://colab.sandbox.google.com/gist/SuryanarayanaY/bd9c2a64c4679c50b51d9bd8fb1eadef/keras3-vit_small_ds-sample-gist.ipynb) for reference.

May please review and suggest any more changes still needed?

### CODE REVIEW

File: examples/vision/vit_small_ds.py [hunk 1]
```diff
@@ -34,30 +34,34 @@
 example is inspired from
 [Image classification with Vision Transformer](https://keras.io/examples/vision/image_classification_with_vision_transformer/).
 
-_Note_: This example requires TensorFlow 2.6 or higher, as well as
-[TensorFlow Addons](https://www.tensorflow.org/addons), which can be
-installed using the following command:
+_Note_: This example requires TensorFlow 2.6 or higher
```
Comment (fchollet) [inline examples/vision/vit_small_ds.py:37]: You can remove this note since it is now inaccurate.

File: examples/vision/vit_small_ds.py [hunk 2]
```diff
@@ -34,30 +34,34 @@
...
-```python
-pip install -qq -U tensorflow-addons
-```
 """
 """
 ## Setup
 """
+import os
+
+os.environ["KERAS_BACKEND"] = "tensorflow"
```
Comment (fchollet) [inline examples/vision/vit_small_ds.py:45]: Would it work with any other backend?

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: examples/vision/vit_small_ds.py (hunk 2 above)
Comment (SuryanarayanaY) [reply examples/vision/vit_small_ds.py:45]: We have dependencies on TF in the below imports.
```
from tensorflow import image as tf_image
from tensorflow import range as tf_range
from tensorflow import constant as tf_constant
```

From `tf.image` module this tutorial uses below APIs:
After exploring more, I found keras equivalents of them except for one as below.

> `tf.image.crop_to_bounding_box `--> Can't find equivalent in keras yet
> `pad_to_bounding_box` --> `keras.ops.image.pad_images`
> `tf.image.extract_patches `--> `keras.ops.image.extract_patches`
> `tf.image.resize` --> `keras.ops.image.resize`

I can replace the above 3 Tf APIs with keras equivalents. Need your suggestion for first one.

Also I am unable to find exact kears equivalents of `tf.constant` and `tf.range`.

Tested the tutorial with the above 3 replacements and its working fine. Attached [gist](https://colab.sandbox.google.com/gist/SuryanarayanaY/199b87f25c33b049da44dd9015ecc62c/keras3-vit_small_ds-sample-gist-r1.ipynb) for reference.

If I get Keras equivalents of `tf.image.crop_to_bounding_box `,`tf.constant` and `tf.range` may be we can make this tutorial backend agnostic.

File: examples/vision/vit_small_ds.py (hunk 1 above)
Comment (SuryanarayanaY) [reply examples/vision/vit_small_ds.py:37]: Ok

Comment (SuryanarayanaY) [reply examples/vision/vit_small_ds.py]: Changes done

File: examples/vision/vit_small_ds.py (hunk 2 above)
Comment (fchollet) [reply examples/vision/vit_small_ds.py:44]: Thank you for looking into it!

`crop_to_bounding_box` is just a slice, I believe, so you can probably do:

```python
# Before:
out = tf.image.crop_to_bounding_box(
    image, offset_height, offset_width, target_height, target_width
)

# After:
out = image[:, offset_height:offset_height+target_height, offset_width:offset_width+target_width, :]
```

`tf.constant` is just `keras.ops.array`. `tf.range` is `keras.ops.arange`.

Comment (SuryanarayanaY) [reply examples/vision/vit_small_ds.py]: Thanks for all the inputs.I have removed the TF dependencies and now the tutorial is backend-agnostic. Attched final [gist](https://colab.sandbox.google.com/gist/SuryanarayanaY/16bf67027603fa8ed8517a9433e14a5d/keras3_test_vit_small_ds-final.ipynb#scrollTo=g86R50Zi9jRD) for tested reference.

**Review by fchollet — COMMENTED** [review]
This is great! The code looks good. Please add the generated files (via the command `python autogen.py add_example vision/vit_small_ds`).

**Comment by SuryanarayanaY** [comment]
@fchollet , generated the `.md` and `.ipynb` files. But accuracy seems dropped compared to earlier.
